apiserver/client/status.go

```diff
@@ -589,6 +590,30 @@ type stateAgent interface {
 	Status() (state.Status, string, map[string]interface{}, error)
 }
 
+// processUnitAgent retrieves status information for both unit and unitAgents.
+func processUnitAgent(unit *state.Unit) (agent api.AgentStatus,
+	workload api.AgentStatus,
+	compatStatus params.Status,
+	compatInfo string) {
+
+	unitAgent := unit.Agent().(*state.UnitAgent)
+
+	workload, _, _ = processAgent(unit)
+	processAgentStatus(&agent, unitAgent)
+
+	compatStatus = params.TranslateToLegacyAgentState(workload.Status)
```

Comment:
It looks like TranslateToLegacyAgentState is being called twice - here and above?